internal/command/command.go

```diff
@@ -20,79 +20,87 @@ type Command struct {
 	ID          string   `json:"id"`
 	Command     []string `json:"command"`
 	Interactive bool     `json:"interactive"`
-	Name        string   `json:"name"`
+	DisplayName string   `json:"name"`
 }
 
-type templateInputs struct {
+// TemplateData is the data passed to command templates to render the command arguments.
+type TemplateData struct {
 	LocalPort int
 	Args      Args
 	Outputs   map[string]string
 }
 
-// toCmd returns a golang cmd object from the calling command.
-func (c Command) toCmd(ctx context.Context, config *Config, cmdArgs Args, outputs Outputs) (*exec.Cmd, error) {
-	name, args, err := c.render(config, cmdArgs, outputs, true)
-	if err != nil {
-		return nil, err
-	}
-
-	// nolint:gosec
-	return exec.CommandContext(ctx, name, args...), nil
+// TemplateOptions are the configurable options used in different rendering contexts.
+type TemplateOptions struct {
+	// ShowSensitive indicates whether sensitive values should be shown. When false, sensitive values are replaced with asterisks.
+	ShowSensitive bool
 }
 
-func (c Command) render(config *Config, cmdArgs Args, outputs Outputs, showSensitive bool) (name string, args []string, err error) {
-	name = c.Command[0]
-	rawArgs := []string{}
+// Name returns the name of the program.
+func (c Command) Name() string {
+	return c.Command[0]
+}
 
-	if len(c.Command) > 1 {
-		rawArgs = append(rawArgs, c.Command[1:]...)
+// Args renders the command arguments using the provided template data and options.
+func (c Command) Args(data TemplateData, options TemplateOptions) ([]string, error) {
+	if len(c.Command) <= 1 {
+		return []string{}, nil
 	}
 
-	args = make([]string, len(rawArgs))
+	args := c.Command[1:]
```

Comment:
This sub slice still points to the elements in c.Command, so below on 68 it ends up replacing the c.Command elements with the templated results.